<launch>

    <!-- Start Sim -->
    <node
        pkg="enph353_utils"
        type="run_sim.sh"
        name="simulator"
        output="screen"
        args="-vpgw"
    />

    <!-- Score tracker -->
    <node 
        pkg="enph353_utils"
        type="score_tracker.py"
        name="score_tracker"
        output="screen"
        cwd="node"
    />

    <!-- GUI Node -->
    <node pkg="competition"
        type="gui.py"
        name="competition_gui"
        output="screen"/>

    <!-- BoardDetector inference node -->
    <node pkg="clueboard_detection"
          type="board_detector.py"
          name="board_detector"
          output="screen">

        <param name="model_path" value="/home/fizzer/ENPH-353-COMPETITION/src/clueboard_detection/runs/detect/clueboards_exp12/weights/best.pt"/>

        <!-- Camera topics -->
        <param name="camera_topic_left" value="/B1/left_front_cam/left_front/image_raw"/>
        <param name="camera_topic_right" value="/B1/right_front_cam/right_front/image_raw"/>
    </node>


    <!-- Line follow inference node -->
    <node pkg="line_following"
          type="inference_node.py"
          name="line_follower"
          output="screen">

        <!-- Image, lidar topics -->
        <param name="image_topic" value="/B1/rrbot/camera1/image_raw"/>
        <param name="scan_topic" value="/B1/scan"/>
        <param name="front_scan_topic" value="/B1/front_scan"/>
        <param name="left_scan_topic" value="/B1/left_scan"/>
        <param name="right_scan_topic" value="/B1/right_scan"/>
        <param name="side_left_scan_topic" value="/B1/side_left_scan"/>
        <param name="side_right_scan_topic" value="/B1/side_right_scan"/>

        <param name="cmd_vel_topic" value="/B1/cmd_vel"/>

        <!-- Model -->
        <param name="model_path" value="$(find line_following)/best_model.pth"/>
    </node>

</launch>
